taxonomy/tests/test_api.py

```diff
@@ -0,0 +1,346 @@
+                          lambda: add_term_to_learning_resource(
+                              -3, self.term.id))
+        self.assertRaises(Term.DoesNotExist,
+                          lambda: add_term_to_learning_resource(
+                              self.learning_resource.id, -4))
+        self.assertRaises(LearningResource.DoesNotExist,
+                          lambda: add_term_to_learning_resource(
+                              None, None))
+
+    def test_add_type_for_vocabulary(self):
+        """
+        Test add_type_for_vocabulary
+        """
+        self.assertEquals(0, self.vocabulary.learning_resource_types.count())
+        add_type_for_vocabulary(self.learning_resource_type1.name,
+                                self.vocabulary.id)
+        self.assertEquals([self.learning_resource_type1],
+                          list(self.vocabulary.learning_resource_types.all()))
+        add_type_for_vocabulary(self.learning_resource_type2.name,
+                                self.vocabulary.id)
+        self.assertEquals([self.learning_resource_type1,
+                           self.learning_resource_type2],
+                          list(
+                              self.vocabulary.learning_resource_types.
+                              order_by('name')))
+
+        # duplicate is ignored
+        add_type_for_vocabulary(self.learning_resource_type1.name,
+                                self.vocabulary.id)
+        self.assertEquals([self.learning_resource_type1,
+                           self.learning_resource_type2],
+                          list(
+                              self.vocabulary.learning_resource_types.
+                              order_by('name')))
+
+    def test_create_term(self):
+        """
+        Test create_term
+        """
+        self.assertEquals([self.term],
+                          list(Term.objects.all()))
+
+        term = create_term(self.vocabulary.id, "label", 9)
+        self.assertEquals([self.term, term],
+                          list(Term.objects.order_by('id')))
+
+        self.assertRaises(Vocabulary.DoesNotExist,
+                          lambda: create_term(456, "label", 9))
+        self.assertRaises(IntegrityError,
+                          lambda: create_term(self.vocabulary.id, None, 9))
+        create_term(self.vocabulary.id, "label2", 99.7)
+        # term weight is truncated after being stored in database
+        self.assertEquals(1, Term.objects.filter(weight=99).count())
+
+    def test_create_vocabulary(self):
+        """
+        Test create_vocabulary
+        """
+        self.assertEquals([self.vocabulary],
+                          list(Vocabulary.objects.all()))
+
+        v = create_vocabulary(self.repository.id,
+                              "vocabulary",
+                              "description",
+                              False,
+                              "f",
+                              9)
```

Comment:
It could be just me, but I think all the `"f"` and `"m"` values here should be replaced with variables at the top of the function. Maybe even with constants defined in models.py and used within the field `choices()` and imported here (and elsewhere) when creating instances.